src/angularjs/src/app/components/scoremetadata.service.js

```diff
@@ -8,6 +8,47 @@
 (function() {
     'use strict';
 
+    /**
+     Sort metrics in metadata by category, with total first in the category.
+     Add subscoreClass property to metrics that should be indented under the total.
+     */
+    function orderCategories(response) {
```

Comment:
Yes, the two goals are definitely in contention. You cannot have arbitrary ordering at the same time as indentation for subcategories, as that implies at least some constraints on order. I can't find the comment, but I did mention this before. I think for now we can do as you suggest and "rely on a stable sort order from the API that matches [the assumption that the order is constrained by the subcategories all appearing directly after their associated top-level category]". 

In any case, we'd need to revisit this with any significant revision to the sort order, and I'm not worried about that happening ATM. 

I don't believe there are any other uses of category in the public site.